buffer/src/main/java/io/netty/buffer/PoolArena.java

```diff
@@ -328,25 +641,56 @@ public synchronized String toString() {
                 }
             }
         }
-        buf.append(StringUtil.NEWLINE);
+    }
 
-        return buf.toString();
+    @Override
+    protected final void finalize() throws Throwable {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Do not override the Object.finalize() method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AObjectFinalizeOverridenCheck)